server/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -315,9 +315,9 @@ private static ObjectType merge(ObjectType left, ObjectType right) {
     public static ObjectType merge(ObjectType left,
                                    ObjectType right,
                                    BiFunction<DataType<?>, DataType<?>, DataType<?>> remappingFunction) {
-        ObjectType.Builder mergedObjectBuilder = ObjectType.of(ColumnPolicy.DYNAMIC);
+        ObjectType.Builder mergedObjectBuilder = ObjectType.of(right.columnPolicy);
```

Comment:
To my understanding, this will affect `union` outputs, do we want the policies from RHS to take precedence?

Reply:
Right, thanks for mentioning. As the right inner types will override the left inner types on naming collision already, I followed this pattern also for the policy. Do you see any issue with this especially for any known union case?

Reply:
Currently we union different types according to the precedence:
https://cratedb.com/docs/crate/reference/en/latest/general/dql/union.html#union-of-different-types
>If types do not match, CrateDB will choose one of the type with higher precedence and try to implicitly cast the values with lower precedence.

I do not have a concrete example that we should not make such change, just thought it is a bit inconsistent and keeping them as-is(default dynamic policy) would be better. But no strong opinion here.

Reply:
> Currently we union different types according to the precedence:

Yes, but objects have the same precedence. In this case, inner types are overwritten by the right object, already.

Reply:
I've pushed a fixup which will not apply the right's column policy on Unions.